internal/cmdrpkgfindupdates/command.go

```diff
@@ -0,0 +1,221 @@
+	"github.com/GoogleContainerTools/kpt/internal/pkg"
+	"github.com/GoogleContainerTools/kpt/internal/util/porch"
+	"github.com/GoogleContainerTools/kpt/internal/util/rpkgutil"
+	kptfile "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1"
+	porchapi "github.com/GoogleContainerTools/kpt/porch/api/porch/v1alpha1"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	command = "cmdrpkgfindupdates"
+)
+
+func NewCommand(ctx context.Context, rcg *genericclioptions.ConfigFlags) *cobra.Command {
+	return newRunner(ctx, rcg).Command
+}
+
+func newRunner(ctx context.Context, rcg *genericclioptions.ConfigFlags) *runner {
+	r := &runner{
+		ctx:    ctx,
+		cfg:    rcg,
+		client: nil,
+	}
+
+	c := &cobra.Command{
+		Use: "find-updates PACKAGE_REVISION",
+		//Short:   rpkgdocs.FindUpdatesShort,
+		//Long:    rpkgdocs.FindUpdatesShort + "\n" + rpkgdocs.FindUpdatesLong,
+		//Example: rpkgdocs.FindUpdatesExamples,
+		PreRunE: r.preRunE,
+		RunE:    r.runE,
+		Hidden:  porch.HidePorchCommands,
+	}
+	r.Command = c
+	return r
+}
+
+type runner struct {
+	ctx     context.Context
+	cfg     *genericclioptions.ConfigFlags
+	client  client.Client
+	Command *cobra.Command
+
+	// Flags
+}
+
+func (r *runner) preRunE(_ *cobra.Command, args []string) error {
+	const op errors.Op = command + ".preRunE"
+
+	if len(args) < 1 {
+		return errors.E(op, "PACKAGE_REVISION is a required positional argument")
+	}
+	cf, err := r.cfg.ToRESTConfig()
+	if err != nil {
+		return errors.E(op, err)
+	}
+	scheme, err := rpkgutil.CreateScheme()
+	if err != nil {
+		return errors.E(op, err)
+	}
+	c, err := client.New(cf, client.Options{Scheme: scheme})
+	if err != nil {
+		return errors.E(op, err)
+	}
+	r.client = c
+	return nil
+}
+
+func (r *runner) runE(cmd *cobra.Command, args []string) error {
+	const op errors.Op = command + ".runE"
+	var messages []string
+
+	for _, p := range args {
+		packageRevision, err := r.getPackageRevision(p)
+		if err != nil {
+			return errors.E(op, err)
+		}
+
+		upstreamLock, err := r.getUpstreamLock(packageRevision)
+		if err != nil {
+			return errors.E(op, err)
+		}
+
+		availableUpdates, err := r.availableUpdates(upstreamLock)
+		if err != nil {
+			return errors.E(op, err)
+		}
+
+		fmt.Fprintf(cmd.OutOrStdout(), "%s %s\n", p, availableUpdates)
+	}
+
+	if len(messages) > 0 {
+		return errors.E(op, fmt.Errorf("errors:\n  %s", strings.Join(messages, "\n  ")))
+	}
+
+	return nil
+}
+
+func (r *runner) getPackageRevision(p string) (porchapi.PackageRevision, error) {
+	packageRevision := porchapi.PackageRevision{}
+	newScheme := runtime.NewScheme()
+	if err := porchapi.SchemeBuilder.AddToScheme(newScheme); err != nil {
+		return packageRevision, err
+	}
+	restClient, err := porch.CreateRESTClient(r.cfg)
```

Comment:
Nit: We can probably construct this one in `preRunE`.

Reply:
I have changed this to be `r.client` and constructed in `preRunE`